src/crypto/crypto.js

```diff
@@ -266,10 +295,19 @@ export function parseEncSessionKeyParams(algo, bytes) {
       const C = new ECDHSymkey(); C.read(bytes.subarray(read));
       return { V, C };
     }
+    //   Algorithm-Specific Fields for X25519 encrypted session keys:
+    //       - 32 octets representing an ephemeral X25519 public key.
+    //       - A one-octet size of the following fields.
+    //       - The one-octet algorithm identifier, if it was passed (in the case of a v3 PKESK packet).
+    //       - The encrypted session key.
     case enums.publicKey.x25519: {
       const ephemeralPublicKey = bytes.subarray(read, read + 32); read += ephemeralPublicKey.length;
-      const C = new ECDHSymkey(); C.read(bytes.subarray(read));
-      return { ephemeralPublicKey, C };
+      let followLength = bytes[read++];
+      const metadata = version === 3 ?
```

Comment:
Thanks for the input. The length is included because `serializeParams` has to write it without any further logic. 
Thus, if the length is not included serialization will fail.

Reply:
I see, did you also discuss the option of changing `serializeParams` to take care of the new 25519/448 params manually? Namely something like:
```js
switch(algo) {
  case enums.publicKey.x25519:
  case enums.publicKey.ed25519:
    const serializedC =  params.C.write();
    return util.concatUint8Arrays([
       params.ephemeralPublicKey,
       [serializedC.length += (params.symmetricAlgo !== null)]
       serializedC
    ]);
  default:
    <same as now>
}
```

I think it could make sense, in order to make this `metadata` handling more readable, esp. considering that it'd otherwise rely on `algosWithNativeRepresentation`, which was originally intended to just mark numbers that shouldn't be serialized as MPIs.

Reply:
Yes, I think this would make it more readable. I will refactor it.